### PR TITLE
docs: add more docs, update README to match (future) landing page content

### PR DESCRIPTION
**Pull request by mudler**
Updates docs to match landing page content

### CODE REVIEW

**Comment by mudler** [comment]
I have some other changes flying for #92 , will rebase on top of this one

**Comment by mudler** [comment]
Just more polishment is left, but I think it's good for a first review

**Comment by mudler** [comment]
@davidcassany I've also added a small paragraph about `SELinux` support, not sure if it's enough/allright, but should be a good start

**Review by davidcassany — APPROVED** [review]
I did a first quick read and it all looks good, just one comment regarding the SELinux

**Review by Itxaka — APPROVED** [review]

**Comment by davidcassany** [comment]
@mudler @Itxaka 

I have been thinking quite a bit about the story we could tell regarding cOS and the toolkit, just writing it here let me know what you think.

Beyond the toolkit concept (how we describe and build it and how we can easily iterate) I believe the cool thing we are providing with this approach is taking the development & deploy workflow of containers to baremetal boxes (or VMs which is the same in that context). 

Upgrading could be as simple as `clonalctl pull` (being `clonalctl` the client of for the `clonal` project or whatever name we use at the end ;) ) and we also keep around the old image in case the new image from `clonalctl pull` does not work as expected. Then as well as with container we could just pull any (valid) appliance we have in the repository, in the same infra and we could easily test different appliances (epinio, k3s, harvester, some non k8s related utility, etc.) without much interaction.

So that brings me to the point that what cOS is really providing is the (i) operation workflow (a client to deploy appliances in a docker like fashion), (ii) a definition/good practices about how to define valid appliances (immutability approach) and (iii) the booting/recovery layout on the system (the very basic and minimalistic system we provide so you can always rollback).

I'd say what we are doing now **is** the recovery partition plus the bootloader, nothing else. I would even divide the recovery image from the appliance, meaning I would consider splitting these two, for derivative (in general) it should not be needed to modify the recovery partition as this should be only used for trouble shooting, no production services should live there IMHO. I would create a damn simple image (as we are doing) and call it the recovery partition and then a derivative system would just provide the contents for COS_ACTIVE/COS_PASSIVE content, all the rest it the plataform we provide to operate on top of it.

So in general I would use three assets for creating an image:
  * bootloader -> provided by us
  * recovery system -> provided by us
  * appliance -> any user defined and highly customized image
  
In fact we could even deliver images without any appliance and just fill it on first boot or filled with a dummy system used to simply pull something else on runtime. So derivate users should not even need to create disk images at all, they should just download our vanilla image for the right platform (public cloud, vmware, baremetal, etc.) and then just deploy their own appliances on it. So image building should be for advanced users only and it should be completely out of the scope at user entry level. I believe this is a relevant point to simplify its use and its adoption.

**Comment by mudler** [comment]
> @mudler @Itxaka
> 
> I have been thinking quite a bit about the story we could tell regarding cOS and the toolkit, just writing it here let me know what you think.
> 
> Beyond the toolkit concept (how we describe and build it and how we can easily iterate) I believe the cool thing we are providing with this approach is taking the development & deploy workflow of containers to baremetal boxes (or VMs which is the same in that context).
> 
> Upgrading could be as simple as `clonalctl pull` (being `clonalctl` the client of for the `clonal` project or whatever name we use at the end ;) ) and we also keep around the old image in case the new image from `clonalctl pull` does not work as expected. Then as well as with container we could just pull any (valid) appliance we have in the repository, in the same infra and we could easily test different appliances (epinio, k3s, harvester, some non k8s related utility, etc.) without much interaction.

Yes definetly, I really hope we get there with #108 so we can offer a better UX there :)

> 
> So that brings me to the point that what cOS is really providing is the (i) operation workflow (a client to deploy appliances in a docker like fashion), (ii) a definition/good practices about how to define valid appliances (immutability approach) and (iii) the booting/recovery layout on the system (the very basic and minimalistic system we provide so you can always rollback).
> 
> I'd say what we are doing now **is** the recovery partition plus the bootloader, nothing else. I would even divide the recovery image from the appliance, meaning I would consider splitting these two, for derivative (in general) it should not be needed to modify the recovery partition as this should be only used for trouble shooting, no production services should live there IMHO. I would create a damn simple image (as we are doing) and call it the recovery partition and then a derivative system would just provide the contents for COS_ACTIVE/COS_PASSIVE content, all the rest it the plataform we provide to operate on top of it.
> 
> So in general I would use three assets for creating an image:
> 
>     * bootloader -> provided by us
> 
>     * recovery system -> provided by us
> 
>     * appliance -> any user defined and highly customized image
> 
> 
> In fact we could even deliver images without any appliance and just fill it on first boot or filled with a dummy system used to simply pull something else on runtime. So derivate users should not even need to create disk images at all, they should just download our vanilla image for the right platform (public cloud, vmware, baremetal, etc.) and then just deploy their own appliances on it. So image building should be for advanced users only and it should be completely out of the scope at user entry level. I believe this is a relevant point to simplify its use and its adoption.

While I do agree fully with you and this is already technically doable, as a toolkit we should empower the user to build their own more than offer a derivative by itself - at the end what you describe here can be already achieved by using the ISOs from our CI, and then later on upgrade to a different image.

I do see the point of separating and we could indeed offer a parameter during install to achieve this (e.g. specify separate recovery image during creation, and so on)

File: docs/derivatives_featureset.md [hunk 1]
```diff
@@ -201,17 +243,30 @@ By default, `cOS` ships a set of default configurations which can be found under
...
 
+[See also the example appliance](https://github.com/rancher-sandbox/epinio-appliance-demo-sample#images)
+
+## SELinux policy
 
+By default, derivatives have `SELinux` enabled in permissive mode. You can use the [cos-toolkit](https://github.com/rancher-sandbox/cOS-toolkit/tree/master/packages/selinux-policies) default policy as a kickstart to customize on top. 
+
+Copy the package (create a new folder with `build.yaml`, `definition.yaml` and `cOS.te`) into the derivative tree and customize to suit your needs, and add it as a build requirement to your OS package.
+
+See also [debian docs on how to write custom policies](https://debian-handbook.info/browse/en-US/stable/sect.selinux.html)
```
[the remark this answers is not in the document]
Comment (mudler) [reply docs/derivatives_featureset.md]: Sounds good! added :+1:

**Comment by davidcassany** [comment]
@mudler @Itxaka 

I have been thinking quite a bit about the story we could tell regarding cOS and the toolkit, just writing it here let me know what you think.

Beyond the toolkit concept (how we describe and build it and how we can easily iterate) I believe the cool thing we are providing with this approach is taking the development & deploy workflow of containers to baremetal boxes (or VMs which is the same in that context). 

Upgrading could be as simple as `clonalctl pull` (being `clonalctl` the client of for the `clonal` project or whatever name we use at the end ;) ) and we also keep around the old image in case the new image from `clonalctl pull` does not work as expected. Then as well as with container we could just pull any (valid) appliance we have in the repository, in the same infra and we could easily test different appliances (epinio, k3s, harvester, some non k8s related utility, etc.) without much interaction.

So that brings me to the point that what cOS is really providing is the (i) operation workflow (a client to deploy appliances in a docker like fashion), (ii) a definition/good practices about how to define valid appliances (immutability approach) and (iii) the booting/recovery layout on the system (the very basic and minimalistic system we provide so you can always rollback).

I'd say what we are doing now **is** the recovery partition plus the bootloader, nothing else. I would even divide the recovery image from the appliance, meaning I would consider splitting these two, for derivative (in general) it should not be needed to modify the recovery partition as this should be only used for trouble shooting, no production services should live there IMHO. I would create a damn simple image (as we are doing) and call it the recovery partition and then a derivative system would just provide the contents for COS_ACTIVE/COS_PASSIVE content, all the rest it the plataform we provide to operate on top of it.

So in general I would use three assets for creating an image:
  * bootloader -> provided by us
  * recovery system -> provided by us
  * appliance -> any user defined and highly customized image
  
In fact we could even deliver images without any appliance and just fill it on first boot or filled with a dummy system used to simply pull something else on runtime. So derivate users should not even need to create disk images at all, they should just download our vanilla image for the right platform (public cloud, vmware, baremetal, etc.) and then just deploy their own appliances on it. So image building should be for advanced users only and it should be completely out of the scope at user entry level. I believe this is a relevant point to simplify its use and its adoption.

> as a toolkit we should empower the user to build their own more than offer a derivative by itself - at the end what you describe here can be already achieved by using the ISOs from our CI, and then later on upgrade to a different image.

Yes sure, my point is that it would be really nice IMHO that users to start testing their own services/appliances build using the toolkit, they only need to build a package like `system/cos` (or on top of `system/cos`) , publish it somewhere and run on of our off the shelf images with a proper yaml referring to just created & published luet package. Of course we would still provide all the tools and workflows to build from scratch if the user requires customization on the layout or needs some other super specific stuff.
 
> I do see the point of separating and we could indeed offer a parameter during install to achieve this (e.g. specify separate recovery image during creation, and so on)

Yes that would be nice. Gonna create a card for the recovery image concept

**Pull request merged**